code/hw_5_pandas.py

```diff
@@ -0,0 +1,228 @@
+# -*- coding: utf-8 -*-
+"""
+This is just a copy from gcollab, a pale decoy, boring fake
+"""
+""" 1 """
+import pandas as pd
+
+
+def read_gff(annotation_file: str):
+    col_names = ['chromosome', 'source', 'type', 'start', 'end', 'score', 'strand', 'phase', 'attributes']
+    annotation_df = pd.read_csv(annotation_file, sep='\t', comment='#', names=col_names)
+
+    return annotation_df
+
+
+def read_bed6(alignment_file: str):
+    col_names = ['chromosome', 'start', 'end', 'name', 'score', 'strand']
+    alignment_df = pd.read_csv(alignment_file, sep='\t', comment='#', names=col_names)
+
+    return alignment_df
+
+
+def check_intersection(start_x, end_x, start_y, end_y): # checks if a contig completely intersects with some rRNA gene
+    if start_y <= start_x and end_y >= end_x:
+        return 'YES'
+    else:
+        return 'NOPE'
+
+
+if __name__ == '__main__':
+    rrna_df = read_gff('rrna_annotation.gff')
+    lines_to_replace = ['Name=16S.*$', 'Name=23S.*$$', 'Name=5S.*$']
+    rrna_df['attributes'] = rrna_df['attributes'].replace(lines_to_replace, ['16S', '23S', '5S'], regex=True)
+
+    rrna_count = rrna_df.groupby('chromosome')['attributes'].value_counts() # counts rRNA in each chromosome
+    sum_df = rrna_count.to_frame(name='quantity').reset_index(level=['chromosome', 'attributes']) # creates a data frame
+    
+    sum_df = sum_df.pivot(index='chromosome', columns='attributes', values='quantity')  # reshapes the data frame for a nice barplot
+    
+    rRNA_barplot = sum_df.plot.bar(figsize=(15, 10), xlabel='',
+                                   title='rRNA distribution in reference genomes',
+                                   subplots=True)
+  
+    align_df = read_bed6('alignment.bed')
+
+    ann_align_df = rrna_df.merge(align_df, on='chromosome') # merges two data frames
+
+    ann_align_df['result'] = ann_align_df.apply(lambda row: check_intersection(row['start_x'], row['end_x'], 
+                                                                           row['start_y'], row['end_y']), axis=1)
+
+    complete_rRNA_df = ann_align_df.loc[ann_align_df['result'] == 'YES'].drop('result', axis=1) # creates a data frame with complete rRNAs sequences

```

Comment:
Можно сделать проще и без вспомогательных функций 
```suggestion
    complete_rRNA_df = ann_align_df.query("start_x >= start_y and end_x <= end_y")

```